spec/lib/booking_requests/booking_requests_spec.rb

```diff
@@ -2,8 +2,16 @@
   let(:api) { instance_double(BookingRequests::Api) }
 
   describe '.slots' do
-    before do
+    around do |example|
+      previous_api = BookingRequests.api
       BookingRequests.api = api
+
+      example.run
+
+      BookingRequests.api = previous_api
```

Comment:
I think for this to be guaranteed it needs to be in an `ensure` since the example could fail but others still run after, so this state would leak. Also, it's duplicated below, maybe we can extract it?